### PR TITLE
[xbmc] Remove the use of obsoleted random_shuffle and move to the new shuffl…

### PR DESCRIPTION
**Pull request by Paxxi**
…e and random framework

### CODE REVIEW

**Comment by Paxxi** [comment]
I can say right now that I will ignore any comments about random_device is slow because these operations isn't something we do often or basically never and not in any performance sensitive parts.
jenkins build this please

**Comment by Montellese** [comment]
Could you split the include reshuffling from the actual code changes?

And maybe it would make sense to write a template because you have duplicated the same code 5 times with slightly different input:
```cpp
template<class TIterator>
void RandomShuffle(TIterator& begin, TIterator& end)
{
    std::random_device rd;
    std::mt19937 mt(rd());
    std::shuffle(begin, end, mt);
}
```

**Comment by Paxxi** [comment]
I was thinking about it but then there's the question of where to put it and I'm note sure it's worth the extra stuff. I'll revert the include stuff :)

**Comment by Montellese** [comment]
IMO getting rid of code duplication is always worth it. Just add a `utils/Random.h` or something like that and use it from there.

**Comment by Paxxi** [comment]
right so I reworked it and just copy pasted your example :)

File: xbmc/utils/Random.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with XBMC; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
```
Comment (Montellese) [inline xbmc/utils/Random.h]: Odd indentation :-P

**Comment by Montellese** [comment]
And it worked? ;-) Thanks for refactoring.

**Comment by Paxxi** [comment]
yeah it worked :) fixed the silly formatting, it ignores the indentation when pasting it from a snippet in VS it seems, maybe I'm just doing something wrong.

**Comment by Montellese** [comment]
Happens to me all the time in VS when I copy the header as well.

**Comment by Paxxi** [comment]
@mkortstiege xcode project please :)

**Comment by Paxxi** [comment]
jenkins build and merge this please

**Comment by Montellese** [comment]
I aborted the build. Please squash the commits because @mkortstiege commit also contains a fix for the template method.

**Comment by Paxxi** [comment]
ok then, squished :)
jenkins build and merge this please

**Comment by Montellese** [comment]
I'm not cutting you any slack just because we're at devcon ;-)

**Comment by Paxxi** [comment]
no you shouldn't :) did you kill Jenkins or did it die on it's own?

**Comment by Paxxi** [comment]
jenkins build this please

**Comment by Paxxi** [comment]
I'll squish this when it succeeds, trying to avoid Jenkins issues
jenkins build this please

**Comment by Razzeee** [comment]
Damn, why have we forgotten about this? :/

**Comment by MartijnKaijser** [comment]
Because @Paxxi partied to hard?

**Comment by Paxxi** [comment]
Because the stupid android stdlib doesn't implement it and I haven't had time to mess with it

**Comment by Paxxi** [comment]
jenkins build this please

**Pull request merged**